src/egon/data/datasets/power_plants/pv_rooftop_buildings.py

```diff
@@ -0,0 +1,2433 @@
+        """
+        SELECT bus_id, geom
+        FROM grid.egon_mv_grid_district
+        ORDER BY bus_id
+        """,
+        index_col="bus_id",
+        geom_col="geom",
+        epsg=epsg,
+    )
+
+    gdf.index = gdf.index.astype(int)
+
+    logger.debug("Grid districts loaded.")
+
+    return gdf
+
+
+def scenario_data(
+    component: str = "generator",
+    carrier: str = "solar_rooftop",
+    scenario: str = "eGon2035",
+) -> pd.DataFrame:
+    """
+    Get scenario capacity data from eGo^n Database.
+    Parameters
+    -----------
+    component : str
+        Component type to filter table by.
+    carrier : str
+        Carrier type to filter table by.
+    scenario : str
+        Scenario to filter table by.
+    Returns
+    -------
+    geopandas.GeoDataFrame
+        GeoDataFrame with scenario capacity data in GW.
+    """
+    with db.session_scope() as session:
+        query = session.query(EgonScenarioCapacities).filter(
+            EgonScenarioCapacities.component == component,
+            EgonScenarioCapacities.carrier == carrier,
+            EgonScenarioCapacities.scenario_name == scenario,
+        )
+
+    df = pd.read_sql(
+        query.statement, query.session.bind, index_col="index"
+    ).sort_index()
+
+    logger.debug("Scenario capacity data loaded.")
+
+    return df
+
+
+class Vg250Lan(Base):
+    __tablename__ = "vg250_lan"
+    __table_args__ = {"schema": "boundaries"}
+
+    id = Column(BigInteger, primary_key=True, index=True)
+    ade = Column(BigInteger)
+    gf = Column(BigInteger)
+    bsg = Column(BigInteger)
+    ars = Column(String)
+    ags = Column(String)
+    sdv_ars = Column(String)
+    gen = Column(String)
+    bez = Column(String)
+    ibz = Column(BigInteger)
+    bem = Column(String)
+    nbd = Column(String)
+    sn_l = Column(String)
+    sn_r = Column(String)
+    sn_k = Column(String)
+    sn_v1 = Column(String)
+    sn_v2 = Column(String)
+    sn_g = Column(String)
+    fk_s3 = Column(String)
+    nuts = Column(String)
+    ars_0 = Column(String)
+    ags_0 = Column(String)
+    wsk = Column(String)
+    debkg_id = Column(String)
+    rs = Column(String)
+    sdv_rs = Column(String)
+    rs_0 = Column(String)
+    geometry = Column(Geometry(srid=EPSG), index=True)
+
+
+def federal_state_data(to_crs: CRS) -> gpd.GeoDataFrame:
+    """
+    Get feder state data from eGo^n Database.
+    Parameters
+    -----------
+    to_crs : pyproj.crs.crs.CRS
+        CRS to transform geometries to.
+    Returns
+    -------
+    geopandas.GeoDataFrame
+        GeoDataFrame with federal state data.
+    """
+    with db.session_scope() as session:
+        query = session.query(
+            Vg250Lan.id, Vg250Lan.nuts, Vg250Lan.geometry.label("geom")
+        )
+
+    gdf = gpd.read_postgis(
+        query.statement, query.session.bind, index_col="id"
+    ).to_crs(to_crs)
+
+    logger.debug("Federal State data loaded.")
+
+    return gdf
+
+
+def overlay_grid_districts_with_counties(
```

Comment:
Can't you use the existing mapping table `boundaries.egon_map_mvgriddistrict_vg250` for this?

Reply:
As far as I see it `boundaries.egon_map_mvgriddistrict_vg250` only provides the mapping `bus_id` -> `vg250_lan` but not the intersecting geometries which get calculated within `overlay_grid_districts_with_counties`. As I need the geometries within the next steps I can't replace this with this table.